### ISSUE TITLE
"Alphabetical" sorting is not alphabetical - Sorted by ASCII code instead

### ISSUE THREAD

**Issue by anobit** [go-gitea/gitea#15582]
- Gitea version (or commit ref): 1.15.0+dev-171-gf719ffc78
- Git version: Not relevant
- Operating system:
  - Not relevant
- Database (use `[x]`):
  - [X] sqlite3
- Can you reproduce the bug at https://try.gitea.io:
  - [X] Yes (https://try.gitea.io/AnoBit?sort=alphabetically)
  - [ ] No
- Log gist: 
  - Not relevant

## Description
The alphabetical sorting is currently not alphabetical, as lower- and uppercase letters are "sorted" seperately. This is likely due to ASCII Code comparison (e.g. "B = 66", "a" = 97).
In the linked example i made three repositories to showcase this: Beta and Omega, starting both with uppercase letters and alpha, starting with a lowercase. With an alphabetical sorting it would be expected to be sorted as "alpha", "Beta", "Omega". Instead, "alpha" is pushed to the back.

Among other things, this can be an error source for people migrating old, inconsistently named VCS to Gitea.

## Screenshots
![image](https://user-images.githubusercontent.com/19514837/115679451-e2a45b80-a352-11eb-8cf7-81772c9703c4.png)

**Comment by lafriks** [comment]
What database are you using?

**Comment by anobit** [comment]
On the system i migrate to sqllite3. Which one the try.gitea.io instance uses, i do not know.

**Comment by lafriks** [comment]
Yes, this is how sqlite does sorting, so it is only sqlite speciffic problem. To have case insensitive sorting sqlite needs `COLLATE NOCASE ASC` but I don't know if xorm supports that
cc @lunny

**Comment by lunny** [comment]
Just use `OrderBy("name COLLATE NOCASE ASC")` if it's sqlite

**Comment by KaKi87** [comment]
Hello,

I don't know about repo names, but the same issue is present for repo file names.

Thanks

**Comment by wxiaoguang** [comment]
I can see 2 problems here.

The first one is for "database collation". I think SQLite is too special for this case, and it doesn't seem worth to make XORM more complex to support the special sort order. It's highly recommended to use some rich-featured databases like MySQL/PgSQL, then you could convert the table/column collation to what you need.


The second one, I think it could be fixed by #29611 

(I think this issue could be closed after #29611  gets merged)

**Comment by wxiaoguang** [comment]
issue problem: repo list: database-related behavior (table/column collation)

----

https://github.com/go-gitea/gitea/issues/15582#issuecomment-1416864350 : but the same issue is present for repo file names: already fixed by #31384

**Comment by KaKi87** [comment]
Forgejo solved this two years ago : https://codeberg.org/forgejo/forgejo/pulls/2522/files

**Comment by wxiaoguang** [comment]
> Forgejo solved this two years ago : https://codeberg.org/forgejo/forgejo/pulls/2522/files

No, Forgejo never fixed this issue's problem.

And, the PR you mentioned was already perfectly fixed in Gitea 2 years ago: Fix natural sort (#31384), while Gitea has much more new features than Forgejo since the hard-fork.

<img width="1264" height="676" alt="Image" src="https://github.com/user-attachments/assets/9eb35274-ab90-4d0e-a81b-e4105a8a4c03" />

**Comment by wxiaoguang** [comment]
> > Forgejo solved this two years ago : https://codeberg.org/forgejo/forgejo/pulls/2522/files
> 
> No, Forgejo never fixed this issue's problem.

By the way, you don't need to behave like some Forgejo maintainers who keep making lies to Gitea. Open your eyes and take a look at the real world and the truths.
